### DISCUSSION TITLE
digraph_getlist(1) in a vim9script def

### DISCUSSION THREAD

**Discussion by kennypete** [vim/vim#16147] [Q&A] [answered]
I've been trying to figure what's up with this as the only place it seems to not work is in a vim9script function:

This, in vimscript, works:

```vim
fu Dg()
  return digraph_getlist(1)[33 : 34]
endfu
echo Dg()  | " echos [['Nb', '#'], ['DO', '$']]
```

And this, in vim9script, works:

```vim
vim9script
echo digraph_getlist(1)[33 : 34]  # echos [['Nb', '#'], ['DO', '$']]
```

But this, in vim9script, fails (with `E1138: Using bool as a number`):

```vim
vim9script
def Dg(): list<list<string>>
  return digraph_getlist(1)[33 : 34]
  # return digraph_getlist(true)[33 : 34]  # similarly, fails
enddef
echo Dg()
```

And this abomination works:

```vim
vim9script
def Dg(): list<list<string>>
  exe 'legacy let g:all = digraph_getlist(1)'
  return g:all[33 : 34]
enddef
echo Dg()  # echos [['Nb', '#'], ['DO', '$']]
```

Am I missing something obvious?

**Comment by habamax** [comment]
> Am I missing something obvious?

Looks like a bug to me.

**Comment by habamax** [comment, marked as answer]
should be fixed now https://github.com/vim/vim/issues/16154

**Reply by kennypete** [reply]
Works great now. Thanks!
